tdp/core/deployment/deployment_plan.py

```diff
@@ -134,41 +138,48 @@ def from_operations(operations: List[Operation]) -> "DeploymentPlan":
     def from_reconfigure(
         dag: Dag,
         cluster_variables: ClusterVariables,
-        component_version_deployed: ComponentVersionLog,
+        deployed_component_version_logs: Iterable[ComponentVersionLog],
     ) -> "DeploymentPlan":
         """Generate a deployment plan based on altered component configuration.
 
         Args:
             dag: DAG to generate the deployment plan from.
             cluster_variables: Cluster variables.
-            component_version_deployed: List of deployed versions.
+            deployed_component_version_logs: List of deployed versions.
 
         Raises:
             RuntimeError: If a service is deployed but the repository is missing.
 
         Returns:
             Deployment plan.
         """
-        components_modified = set()
-        for (
-            service,
-            _component,
-            version,
-        ) in component_version_deployed:
-            if service not in cluster_variables:
-                raise RuntimeError(
-                    f"Service '{service}' is deployed but the repository is missing."
+        operations = set()
+        modified_components = cluster_variables.get_modified_components_names(
+            deployed_component_version_logs,
+        )
+        for modified_component in modified_components:
+            logger.debug(
+                f"Component {modified_component.component_name} of service {modified_component.service_name} has been modified"
+            )
+            is_service = modified_component.component_name is None
+            if is_service:
```

Comment:
You can use the `is_service` method no ?
```suggestion
            if modified_component.is_service():
```